test/unit/OperatorStateRetrieverUnit.t.sol

```diff
@@ -6,8 +6,238 @@ import "../utils/MockAVSDeployer.sol";
+        operatorStateRetriever.getOperatorState(registryCoordinator, BitmapUtils.bitmapToBytesArray(1 << numQuorums), uint32(block.number));
+    }
+
+    function test_getOperatorState_revert_quorumNotCreatedAtReferenceBlockNumber() public {
+        cheats.roll(registrationBlockNumber);
+        IRegistryCoordinator.OperatorSetParam memory operatorSetParams = 
+            IRegistryCoordinator.OperatorSetParam({
+                    maxOperatorCount: defaultMaxOperatorCount,
+                    kickBIPsOfOperatorStake: defaultKickBIPsOfOperatorStake,
+                    kickBIPsOfTotalStake: defaultKickBIPsOfTotalStake
+            });
+        uint96 minimumStake = 1;
+        IStakeRegistry.StrategyParams[] memory strategyParams = new IStakeRegistry.StrategyParams[](1);
+        strategyParams[0] =
+            IStakeRegistry.StrategyParams({
+                strategy: IStrategy(address(1000)),
+                multiplier: 1e16
+            });
+
+        cheats.prank(registryCoordinator.owner());
+        registryCoordinator.createQuorum(operatorSetParams, minimumStake, strategyParams);
+
+        cheats.expectRevert("IndexRegistry._operatorCountAtBlockNumber: quorum did not exist at given block number");
+        operatorStateRetriever.getOperatorState(registryCoordinator, BitmapUtils.bitmapToBytesArray(1 << numQuorums), uint32(registrationBlockNumber - 1));
+    }
+
+    function test_getOperatorState_returnsCorrect() public {
+        uint256 quorumBitmapOne = 1;
+        uint256 quorumBitmapThree = 3;
+        cheats.roll(registrationBlockNumber);
+        _registerOperatorWithCoordinator(defaultOperator, quorumBitmapOne, defaultPubKey);
+
+        address otherOperator = _incrementAddress(defaultOperator, 1);
+        BN254.G1Point memory otherPubKey = BN254.G1Point(1, 2);
+        bytes32 otherOperatorId = BN254.hashG1Point(otherPubKey);
+        _registerOperatorWithCoordinator(otherOperator, quorumBitmapThree, otherPubKey, defaultStake -1);
+
+        OperatorStateRetriever.Operator[][] memory operators = operatorStateRetriever.getOperatorState(registryCoordinator, BitmapUtils.bitmapToBytesArray(quorumBitmapThree), uint32(block.number));
+        assertEq(operators.length, 2, "operators are registered for 2 quorums, so there should be 2 arrays of operators");
+        assertEq(operators[0].length, 2, "operators are registered for 2 quorums, so there should be 1 operator in the first quorum");
+        assertEq(operators[1].length, 1, "operators are registered for 2 quorums, so there should be 1 operator in the second quorum");
+        assertEq(operators[0][0].operatorId, defaultOperatorId, "the first operator in the first quorum should be the default operator");
+        assertEq(operators[0][0].stake, defaultStake, "the first operator in the first quorum should have the default stake");
+        assertEq(operators[0][1].operatorId, otherOperatorId, "the second operator in the first quorum should be the other operator");
```

Comment:
For the `Operator` state checks, could you pull/rebase from m2-mainnet and add operator address checks as well?
I made a small addition to this contract for avs-sync here https://github.com/Layr-Labs/eigenlayer-middleware/pull/143

Reply:
[0d48200](https://github.com/Layr-Labs/eigenlayer-middleware/pull/140/commits/0d48200ca01817b91fa3d062838382973c63ed1b)

Reply:
done! have you checked that getOperatorState still runs within a 50million gas with this change and large operator sets?